src/main/scala/is/hail/stats/SkatModel.scala

```diff
@@ -0,0 +1,140 @@
+package is.hail.stats
+
+import breeze.linalg._
+import com.sun.jna.Native
+import com.sun.jna.ptr.IntByReference
+import is.hail.utils._
+
+/** SKATModel
+  *   Computes the p value of the variance-component score statistic as 
+  *   defined by SKAT.
+  * Input:
+  *   Gw                      (n x m) DenseMatrix
+  *     The genotype matrix for n samples and m genetic variants with each
+  *     column scaled by the corresponding variant weight.
+  *   QtGw                    (k x m) DenseMatrix
+  *     The projection of the weighted genotype matrix Gw by Q.t, where Q
+  *     comes from the reduced QR decomposition of the covariate data.
+  *   qstat                   Double
+  *     Variance-component score statistic at which to evaluate the cumulative
+  *     distribution.
+  * Returns:
+  *   ((qstat, pval), fault)      ((Double, Double), Int)
+  *     SkatStat and the pValue computed by Davies Algorithm. With the statistics
+  *     a integer indicating any issues running Davies Algorithm is also returned.
+  */
+class
+SkatModel(Gw: DenseMatrix[Double], QtGw: DenseMatrix[Double], skatStat: Double) {
+
+  /** Davies Algorithm original C code
+    *   Citation:
+    *     Davies, Robert B. "The distribution of a linear combination of
+    *     x2 random variables." Applied Statistics 29.3 (1980): 323-333.
+    *   Software Link:
+    *     http://www.robertnz.net/QF.htm
+    */
+  @native
+  def qfWrapper(lb1: Array[Double], nc1: Array[Double], n1: Array[Int], r1: Int,
+    sigma: Double, c1: Double, lim1: Int, acc: Double, trace: Array[Double],
+    ifault: IntByReference): Double
+
+  Native.register("hail")
+
+  def computeSkatStats(): ((Double, Double),Int) = {
```

Comment:
make the return type `(SkatStat, Int)` since you're just going to group the first two later